airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -120,6 +120,12 @@ async def skip_backward_compatibility_tests_fixture(
             pytest.skip(f"Backward compatibility tests are disabled for version {previous_connector_version}.")
         return False
 
+    @pytest.fixture(name="skip_oauth_default_method_test")
+    def skip_oauth_default_method_test_fixture(self, inputs: SpecTestConfig):
+        if inputs.auth_default_method and not inputs.auth_default_method.oauth:
```

Comment:
So if `inputs.auth_default_method` is not defined (which seems possible as [it is optional](https://github.com/airbytehq/airbyte/pull/34178/files#diff-6d30b0c7923582524ce6f15dccff0e22689713a162893c6dcd268c6af46f2302R64)), we will assume oauth is the default method, is that right? Is this a fair assumption? I'm afraid that it will force all the connectors to define `inputs.auth_default_method.oauth == False` which would mean that this is a breaking change. I'm not sure how we usually manage breaking changes for CATs. Do you have more context on that?

Reply:
Yes, when `auth_default_method` is not defined in `acceptance-test-config.yml` we assume that OAuth is default method. I'm not sure what you mean about a breaking change. If there is a weigh reason for some connector to do not specify Oauth as default engineer can add `auth_default_method.oauth == False` to cat config. Otherwise, engineer should change the order in oneOf in source connector spec  which is not a breaking change for source.

Reply:
> If there is a weigh reason for some connector to do not specify Oauth as default engineer can add auth_default_method.oauth == False

So will there be a follow up work to fix all the CATs that will fail because of this? For example, source-stripe does not suppor oauth. This change will break CATs for this connector. What is the plan to upgrade all connectors like source-stripe? Or should we make this change non-breaking?

Reply:
according to this comment https://github.com/airbytehq/airbyte/issues/33564#issuecomment-1863272005 I updated existing CAT scenatio, but there is a way to make this change non-breaking: update TestConnectorAttributes class and this test will not work unless config for it will be added to acceptance-test-config.yml.

Reply:
I'll take some time to review this after some morning meetings, but first thoughts on this laying out the possible scenarios:

1. Connector only supports API key -> The test is skipped regardless of CAT config
2. Connector supports OAuth and API key, the OAuth is default -> Passes test
3. Connector supports OAuth and API key, API key is default -> Fails test
4. Connector supports OAuth and API key, API key is default, override is enabled-> Skips/passes test.

> This is a breaking change and if your connector does not support oauth, you need to define inputs.auth_default_method.oauth == False

I think we want to try to avoid this because its pretty involved to change many of our connectors.

Basically, I think we want to get to a place where connectors that don't even support OAuth should not need to define anything in the CAT spec and we skip the test entirely because it's irrelevant. I need to read through this in more detail, but given we have the whole spec, I would think we can figure out if OAuth is one of the auth methods.

And for any source that does have an OAuth as an option, then we need to verify it comes first, unless the override is specified. I think even those this might break existing Silver certified connectors, its a valid test failure because it means we forgot to fix this during the certification